osg/short-term.md

```diff
@@ -0,0 +1,292 @@
+
+### input_ticket_list
+
+This setting contains the name of the input ticket list file.
+
+### output_ticket_list
+
+This setting contains the name of the output ticket list file.
+
+### status_update_url
+
+This setting contains a URL that should be used to send job status updates back to the DE. This will be described in
+more detail later.
+
+### stdout
+
+This setting contains the name of the file that should contain the redirected standard output produced by the tool.
+
+### stderr
+
+This setting contains the name of the file that should contain the redirected standard error output produced by the
+tool.
+
+## Initializing the iRODS Connection
+
+Initializing the iRODS connection is actually fairly simple if you're using version 4.x of the icommands, which is
+highly recommended. The username in the `irods_user` configuration setting is guaranteed to have an empty password, so
+iRODS can be initialized by creating a file called `$HOME/.irods/irods_environment.json`. This file should be in the
+following format:
+
+```
+{
+    "irods_user_name": "<username>",
+    "irods_host": "<host-name>",
+    "irods_port": <port-number>,
+    "irods_zone_name": ""
+}
+```
+
+For the time being, the zone name should always be equal to the empty string. The rest of the settings are extracted
+from the job configuration file. After creating this file, `iget` and `iput` can be used in conjunction with the iRODS
+tickets to retrieve files from or push files to iRODS.
+
+If you're using custom iRODS credentials with a password then (assuming you're not using Jargon) `iinit` must be called
+before using any of the iRODS icommands. `iinit` only runs in interactive mode, so it will be necessary to send the
+password to `iinit` using redirection. Using custom iRODS credentials is expected to be rare, so precise instructions
+for running `iinit` are beyond the scope of this documentation. Similarly, using Jargon is beyond the scope of this
+documentation.
+
+## Sending Job Status Updates
+
+Job status updateS can and should be sent to the DE using HTTP POST requests. The complete URL used to send job status
+updates for the current job is provided in the job configuration file. The request body is a JSON object consisting of
+three fields. The `state` field contains the current job status. The `message` field contains a brief description of
+what the job is currently doing. The `hostname` field contains the host name or IP address tht the request is being sent
+from.
+
+### Status Codes
+
+There are three job status codes that can be used in the wrapper script: `running`, `completed` and `failed`. A status
+update with the status code `running` should be sent immediately after the job configuration file is parsed. This
+informs the user that the job has started. The wrapper script may send as many `running` job status updates as it
+wants. It is often helpful to send job status updates when the wrapper script is about to start downloading files, about
```

Comment:
I think it would be worth specifying that sending these additional job status updates is helpful for debugging, specifically. As it's currently written I'm not sure as an implementor I'd feel like it was worth it, without a more specific reason (and I think we as the devs would really prefer people send too many job status updates, than too few :sweat_smile: )

Reply:
That's a good point. We definitely want to have at least the `completed` or `failed` message.